src/Farmer/Builders/Builders.ResourceGroup.fs

```diff
@@ -5,32 +5,31 @@ open Farmer
 open Farmer.Arm.ResourceGroup
 
 type ResourceGroupConfig =
-    { Name: string Option
+    { Name: ArmExpression
```

Comment:
Ah I understand. The problem we were faced with here is a circular depenedency between hostname binding and certificate.
The certificate depends on the hostNameBinding but the hostNameBinding's SSL state depends on the certficate. As such, it is impossible to deploy a webapp with certfiicate and custom domain all in one deployment. The solution we had for this was to use the nested resource group deployment to provide a second phase in which we update the hostName binding to enable TLS with the certificate.

This can be seen in action in Builders.WebApp.fs Lines 480-512.

It is possible that we could pull out the ResourceGroup name change into a separate PR. However, the rest of this custom domain PR wouldn't be possible until that had been approved.